### DISCUSSION TITLE
Weekly update 2022-06-15

### DISCUSSION THREAD

**Discussion by akhi3030** [near/nearcore#7056] [Contract Runtime]
**Done:**
- @Ekleog: Wrote and landed a small cleanup / API fixing PR: https://github.com/near/nearcore/pull/7013
- @Ekleog: Submitted a PR for differential fuzzing: https://github.com/near/nearcore/pull/7025
- @matklad: https://github.com/near/nearcore/pull/7042, https://github.com/near/nearcore/pull/7043 -- finally dropped some old unused code!
- @matklad: https://github.com/near/nearcore/pull/7034, https://github.com/near/nearcore/pull/6999, https://github.com/near/nearcore/pull/6998, https://github.com/near/nearcore/pull/6967  -- small cleanups
- @nagisa: https://github.com/near/wasmer/pull/135 to delete a grand total of thirty five thousand dead lines of code;
- @nagisa: https://github.com/near/wasmer/pull/128 for linear handling of locals in our compilation pipeline;
- @nagisa: Wrote down a skeleton test framework for instrumentation spec. tests;
- @nagisa: Submitted a draft section of the spec on export indirection as well as add-ons to the base wasm-core-1 runtime semantics spec.
- @jakmeier: Debugged estimator issues. Fixed parts ([nearcore#7002](https://github.com/near/nearcore/pull/7002)) and created issues for remainder ([nearcore#7003](https://github.com/near/nearcore/pull/7003), [nearcore#7004](https://github.com/near/nearcore/pull/7004)).

**Upcoming:**
- @Ekleog: Land https://github.com/near/nearcore/pull/7025
- @matklad: https://github.com/near/nearcore/pull/7038 -- finally starting to do something with actix
- @matklad: https://github.com/near/nearcore/pull/7012 -- further test cleanup, waiting for review
- @matklad: https://github.com/near/nearcore/pull/7044 -- yet another small cleanup
- @nagisa: Writing down findings on mmap performance to document improvements, tradeoffs, etc. This will take a form of a blog post spanning ~8 pages.
- @jakmeier: Spending time understanding IO performance

**Comment by nagisa** [comment]
FWIW the blog post can be found [here](https://kazlauskas.me/entries/fast-page-maps-for-jit)
